### PR TITLE
feat(gossip): data source metrics

### LINKED ISSUE #121: Gossip: Data Source Metrics
right now we just insert new gossip data directly into the table

- would be nice to have metrics to know where our data is being updated from: either from push messages, pull responses, or local 

- would also be nice to know how many crds values are being produced for each pull request received

this would be a new prometheus metric

### PR DESCRIPTION
**Pull request by sonicfromnewyoke**
attempt to close #121

### CODE REVIEW

**Comment by Sobeston** [comment]
Looks all good so far :+1:

File: src/gossip/service.zig [hunk 1]
```diff
@@ -1253,6 +1253,7 @@ pub const GossipService = struct {
                 _ = gossip_table.insert(req.value, now) catch {};
                 gossip_table.updateRecordTimestamp(req.value.id(), now);
             }
+            self.stats.table_n_updates_pull_req.add(pull_requests.items.len);
```
Comment (0xNineteen) [inline src/gossip/service.zig:1256]: `gossip_table.insert` will fail with an error if the value was not inserted which is not accounted for in this case -- would need to add logic to the catch block

File: src/gossip/service.zig [hunk 2]
```diff
@@ -1442,6 +1443,7 @@ pub const GossipService = struct {
                         now,
                     ) catch {};
                 }
+                self.stats.table_n_updates_pull_resp.add(timeout_indexs.items.len);
```
Comment (0xNineteen) [inline src/gossip/service.zig:1446]: same here -- but we need to look at `insertValues` in the lines above

File: src/gossip/service.zig [hunk 3]
```diff
@@ -1769,6 +1771,7 @@ pub const GossipService = struct {
 
         while (push_msg_queue.popOrNull()) |gossip_value| {
             _ = gossip_table.insert(gossip_value, now) catch {};
+            self.stats.table_n_updates_push_msg.inc();
```
Comment (0xNineteen) [inline src/gossip/service.zig:1774]: same here

**Review by 0xNineteen — CHANGES_REQUESTED** [review]
the main thing we want to track is when values are successfully **inserted** and **updated** in two separate metrics

**Comment by 0xNineteen** [comment]
closed for #276

**Pull request closed without merging**